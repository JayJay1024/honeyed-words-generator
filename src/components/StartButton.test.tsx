import React from 'react';
import { create, act, ReactTestRenderer } from 'react-test-renderer';

import StartButton from './StartButton';

describe('component RefreshButton', () => {
    let testRenderer: ReactTestRenderer = null;

    beforeEach(() => {
        act(() => {
            testRenderer = create(<StartButton>testStart</StartButton>);
        });
    });

    afterEach(() => {
        if (testRenderer) {
            testRenderer.unmount();
            testRenderer = null;
        }
    });

    test('RefreshButton should match snapshot', () => {
        let tree = testRenderer.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('there should be a button', () => {
        const testInstance = testRenderer.root;
        expect(testInstance.findAllByType("button").length).toEqual(1);
        expect(testInstance.findByType("button").children[0]).toBe("testStart");
    });

    test('there should be six images', () => {
        const testInstance = testRenderer.root;
        expect(testInstance.findAllByType("img").length).toEqual(6);
    })
});
