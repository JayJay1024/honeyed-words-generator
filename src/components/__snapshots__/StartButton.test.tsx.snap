// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component RefreshButton RefreshButton should match snapshot 1`] = `
<button
  className="sc-AxjAm hQkRzR"
>
  testStart
  <img
    alt="heart"
    className="heart"
    src={Object {}}
    style={
      Object {
        "animationDelay": "0s",
      }
    }
  />
  <img
    alt="heart"
    className="heart"
    src={Object {}}
    style={
      Object {
        "animationDelay": "0.3s",
      }
    }
  />
  <img
    alt="heart"
    className="heart"
    src={Object {}}
    style={
      Object {
        "animationDelay": "0.6s",
      }
    }
  />
  <img
    alt="heart"
    className="heart"
    src={Object {}}
    style={
      Object {
        "animationDelay": "0.8999999999999999s",
      }
    }
  />
  <img
    alt="heart"
    className="heart"
    src={Object {}}
    style={
      Object {
        "animationDelay": "1.2s",
      }
    }
  />
  <img
    alt="heart"
    className="heart"
    src={Object {}}
    style={
      Object {
        "animationDelay": "1.5s",
      }
    }
  />
</button>
`;
